pkg/csi/cinder/controllerserver.go

```diff
@@ -283,14 +284,16 @@ func (cs *controllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
 
 		klog.V(3).Infof("Found existing snapshot %s on %s", name, volumeId)
 	} else if len(snapshots) > 1 {
-		klog.V(3).Infof("found multiple existing snapshots with selected name (%s) during create", name)
-		return nil, errors.New("multiple snapshots reported by Cinder with same name")
+		err := fmt.Sprintf("found multiple existing snapshots with selected name (%s) during create", name)
+		klog.V(3).Infof(err)
+		return nil, status.Error(codes.FailedPrecondition, err)
 	} else {
 		// TODO: Delegate the check to openstack itself and ignore the conflict
 		snap, err = cloud.CreateSnapshot(name, volumeId, description, &req.Parameters)
 		if err != nil {
-			klog.V(3).Infof("Failed to Create snapshot: %v", err)
-			return nil, err
+			errStr := fmt.Sprintf("Failed to Create snapshot: %v", err)
+			klog.V(3).Infof(errStr)
+			return nil, status.Error(codes.Aborted, errStr)
```

Comment:
It should be codes.Internal instead? codes.Aborted Indicates that there is already an operation pending for the specified snapshot which might not be correct in this case?

directly status.Errorf can be used instead. 
`status.Errorf(codes.Internal, "Failed to create snapshot %v", err)`

Reply:
ok, thanks for the info~